string parameter_file_location